test/static_arrays.jl

```diff
@@ -17,13 +17,11 @@ for alg in (nothing, LUFactorization(), SVDFactorization(), CholeskyFactorizatio
     sol = solve(LinearProblem(A, b), alg)
     @inferred solve(LinearProblem(A, b), alg)
     @test norm(A * sol .- b) < 1e-10
-
-    if __non_native_static_array_alg(alg)
-        @test_broken __solve_no_alloc(A, b, alg)
+    if alg isa Function && alg === KrylovJL_GMRES
```

Comment:
```suggestion
    if alg isa KrylovJL{typeof(LinearSolve.Krylov.gmres!)}
```
if you'd prefer `gmres!` only
```julia
julia> alg isa KrylovJL{typeof(LinearSolve.Krylov.gmres!)}
true
```